backends/qualcomm/qnn_quantizer.py

```diff
@@ -226,847 +186,151 @@ def get_ptq_per_channel_weight_config() -> QuantizationConfig:
+    ) -> None:
+        self.custom_quant_annotations = custom_quant_annotations
 
-    def _annotate_cat(self, gm: torch.fx.GraphModule) -> None:
-        op_sources = [torch.ops.aten.cat.default, torch.cat]
-        qconfig = self._get_quant_config(op_sources)
-        cat_partitions = get_source_partitions(gm.graph, op_sources)
+    def add_16bit_quant_ops(self, ops: Set[OpOverload]) -> None:
+        for op in ops:
+            assert (
+                op in self.SUPPORTED_OPS
+            ), f"The annotation of op {op} is not implemented"
 
-        cat_partitions = list(itertools.chain(*cat_partitions.values()))
-        for cat_partition in cat_partitions:
-            cat_node = cat_partition.output_nodes[0]
-            input_nodes = cat_node.args[0]
-            if _is_annotated([cat_node]):
-                continue
+            self.bit8_quant_ops.remove(op)
+            self.bit16_quant_ops.add(op)
 
-            assert isinstance(input_nodes, Sequence)
+    def _get_quant_config(self, op: OpOverload) -> Optional[QuantizationConfig]:
+        """
+        Priority:
+            1. per channel config when enable_per_channel_conv_quant is True
+            2. int8 / int16 config
+        """
+        if self.enable_per_channel_conv_quant and op == torch.ops.aten.conv2d.default:
+            if op in self.bit16_quant_ops:
+                return get_ptq_per_channel_weight_config(torch.int16, torch.int16)
+            return get_ptq_per_channel_weight_config()
 
-            first_input_node = input_nodes[0]
-            input_qspec_map = {}
-            assert isinstance(first_input_node, Node)
-            assert isinstance(cat_node, Node)
-            input_qspec_map[first_input_node] = qconfig.input_activation
-            share_qparams_with_input_act0_qspec = SharedQuantizationSpec(
-                (first_input_node, cat_node)
-            )
+        if op in self.bit8_quant_ops:
+            return self.bit8_quant_config
 
-            for input_node in input_nodes[1:]:
-                if input_node not in input_qspec_map:
-                    assert isinstance(input_node, Node)
-                    input_qspec_map[input_node] = share_qparams_with_input_act0_qspec
+        if op in self.bit16_quant_ops:
+            return self.bit16_quant_config
 
-            cat_node.meta[QUANT_ANNOTATION_KEY] = QuantizationAnnotation(
-                input_qspec_map=input_qspec_map,
-                output_qspec=share_qparams_with_input_act0_qspec,
-                _annotated=True,
-            )
+        raise AssertionError(f"No quant config is implemented for op, {op}")
 
-    def _annotate_upsample2d(self, gm: torch.fx.GraphModule) -> None:
-        op_sources = [torch.nn.functional.interpolate, torch.nn.UpsamplingBilinear2d]
-        qconfig = self._get_quant_config(op_sources)
-        upsample_partitions = get_source_partitions(gm.graph, op_sources)
+    # TODO change to transform_for_annotation from the parent class when it's ready
```

Comment:
I think it's already there https://github.com/pytorch/pytorch/pull/113115

we can have a separate change to handle this todo though

Reply:
Thanks for pointing out! we will prepare another PR for it.